src/codegen/faux-applications.lisp

```diff
@@ -0,0 +1,74 @@
+(defpackage #:coalton-impl/codegen/faux-applications
```

Comment:
I think "faux applications" is maybe not exactly the right term we should use. Maybe "special application" or "intrinsic-application"?